sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -951,6 +951,14 @@ object SQLConf {
     .checkValue(_ > 0, "the value of spark.sql.sources.bucketing.maxBuckets must be greater than 0")
     .createWithDefault(100000)
 
+  val AUTO_BUCKETED_SCAN_ENABLED =
+    buildConf("spark.sql.sources.bucketing.autoBucketedScan.enabled")
+      .doc("When true, decide whether to do bucketed scan on input tables based on query plan " +
+        "automatically.")
+      .version("3.1.0")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
If `spark.sql.sources.bucketing.enabled` is disabled, does this config still work?

Reply:
@viirya - this config will not take any effect in that case, updated.